src/pages/Search.js

```diff
@@ -0,0 +1,103 @@
+    state = {
+        text: '',
+        productList: [
+            {
+                id: 1,
+                name: "Apple",
+                img: [ process.env.PUBLIC_URL + '/images/apple.jpg'],
+                caption: "Keeps the doctor away!",
+                category: "fruit"
+            },
+            {
+                id: 2,
+                name: "Paint brush",
+                img: [ process.env.PUBLIC_URL + '/images/paintbrush.jpg'],
+                caption: "Paint not included.",
+                category: "art"
+            },
+            {
+                id: 3,
+                name: "Cup of coffee",
+                img: [ process.env.PUBLIC_URL + '/images/hotCoffee.jpg'],
+                caption: "Piping hot and full of caffeine!",
+                category: "drink"
+            },
+            {
+                id: 4,
+                name: "Iced coffee",
+                img: [ process.env.PUBLIC_URL + '/images/icedCoffee.jpg'],
+                caption: "Pairs well with our hot coffee.",
+                category: "drink"
+            },
+            {
+                id: 4,
+                name: "Red rose",
+                img: [ process.env.PUBLIC_URL + '/images/redRose.jpeg'],
+                caption: "Smells good!",
+                category: "flower"
+            },
+            {
+                id: 5,
+                name: "Sunflowers",
+                img: [ process.env.PUBLIC_URL + '/images/sunflower.jpg'],
+                caption: "For your garden or kitchen.",
+                category: "flower"
+            }
+        ],
+        filteredProductList: [],
+        showNoResults: false,
+    };
+
+    Search = (searchTerm) => {
+        let filteredProductList = this.state.productList.filter(product =>
+            product.name.toLowerCase().indexOf(searchTerm.toLowerCase()) !== -1);
+        this.setState({filteredProductList: filteredProductList});
+    };
+
+    onSubmit = (e) => {
+        e.preventDefault();
+        this.Search(this.refs.searchForm.value);
+    };
+
+    render() {
+        return (
+            <div>
+                <form onSubmit={this.onSubmit} className="form">
+                    <input
+                        id="searchForm"
+                        ref="searchForm"
+                        type="text"
+                        name="text"
+                        placeholder="Search..."
+                        className="search-bar"
+                    />
+                    <input
+                        type="submit"
+                        value="Search"
+                        className="search-btn"
+                    />
+                </form>
+                <div className="searchResults">
+                    {this.state.filteredProductList.map((product, index) => {
+                        return <div key={index} className="product-profile">
+                            <Link to={`/${product.id}`} className="product-link">
```

Comment:
isn't the route `"/shop/:id"` not `/:id"`?

```<Route exact path="/cart" component={Cart}/>
                    <Route exact path="/shop" component={ShopPage}/>
                    <Route exact path="/shop/:id" component={ProductProfile}/>
                    <Route exact path="/search" component={Search}/>```

Reply:
you are looking at the search component